src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -479,6 +479,7 @@ def load_arguments(self, _):
         c.argument('path', options_list=['--file', '-f'], type=file_type, completer=FilesCompleter(),
                    default=os.path.join(os.path.expanduser('~'), '.kube', 'config'))
         c.argument('public_fqdn', default=False, action='store_true')
+        c.argument('credential_format', type=str, options_list=['--format'], validator=validate_credential_format)
```

Comment:
Also you could use `arg_type=get_enum_type([CONST_1, CONST_2])` instead of the validator.